crates/biome_graphql_analyze/src/lint/nursery/no_lower_case_enum.rs

```diff
@@ -0,0 +1,76 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_graphql_syntax::{GraphqlEnumValueDefinition, GraphqlRoot};
+use biome_rowan::{AstNode, TextRange};
+
+declare_lint_rule! {
+    /// Validates that all enum values are capitalized.
+    ///
+    /// By convention in GraphQL, enum values are all caps.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```graphql,expect_diagnostic
+    /// enum MyEnum {
+    ///  value
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```graphql
+    /// enum MyEnum {
+    ///  VALUE
+    /// }
+    /// ```
+    ///
+    pub NoLowerCaseEnum {
+        version: "next",
+        name: "noLowerCaseEnum",
+        language: "graphql",
+        recommended: false,
+    }
+}
+
+impl Rule for NoLowerCaseEnum {
+    type Query = Ast<GraphqlRoot>;
```

Comment:
Your query should reference the actual node you are looking for, not the root node. So it should be `GraphqlEnumValueDefinition` and ...